### DISCUSSION TITLE
Harmonic signals presence

### DISCUSSION THREAD

**Discussion by iu0azb** [egzumer/uv-k5-firmware-custom#239] [Ideas]
Since I started to use the radio i noticed the presence of false carriers on 130Mhz, 78mhz.. 260 etc. After have deeply investigate I understood that all are harmonic of 26mhz.. the frequency of BK xtal.
Looking at the detailed PCB photo available online I realized that is not used a simple cristal as reported in the datasheet but a xtal generator (Marked AN26.1). If this correspond for example to a DSA221SDN, the 2th 3th 4th harmonic  levels are high.. 
The PCB seems to give room for other configurations... do you think some testing is possible?  
I don't have skills in detail to understand how harmonics on this type of chip would affect reception in general but If it were a classic analog radio I would have already gotten my hands on it :)

![UV-5K 26mhz ](https://github.com/egzumer/uv-k5-firmware-custom/assets/63300383/1b2620a3-8b34-4940-a6f4-99a7ba705a69)

**Comment by Hunterfnet** [comment]
i noticed too near 120-150 Mhz , 18-26 ... etc from local stations broadcast with a lot of power. Is normal.  When i was young and on normal transceivers adjusting on coils or insert one new should solve the nasties interferences os imply open the band .... but this new microscopic technology makes me desist to work with loupe and precision that i have lost for ages, that's the progress , others will implement filters , power , memory and much more , i wish the next generation do something better that using a ChatGPT for ask the 15% of one million.

**Reply by iu0azb** [reply]
The local broadcast can generate intermodutalions and fortunately these can be mitigated with a notch filter betweeen antenna and receier.  I'm mentioning something different, it seem that the xtal generator signal, not very clean, causing the gost carriers along the spectrum.. not a big deal but I/m courios to see if a fix can eventually improve overall the performance. Also is annoying while scanning have the carrier presence that can't bypassed by a lockout funcion yet :)

**Reply by iu0azb** [reply]
Today I've tried to shield a bit the area of xtal. 
![image](https://github.com/egzumer/uv-k5-firmware-custom/assets/63300383/d816aaa2-9da1-4ba6-98c3-d003d6042511)

The result is a reduction of signal of around 10dbm at 130mhz.. not as good as expected 😒

I would attempts to put a passive xtal and see out it goes.. but it's not clear if this configuration need to be set also on software setting.

**Reply by Hunterfnet** [reply]
Maybe the key is change another compatble xtal generator with more quality. I use ijv last firm on other walkie UVK5 for make comparissons. I haven't any internal noticeable interference on my UV5R (external a lot) ,I suppose all units are using the same components ( xtal and board ) . I see too the RF sensitivy changes a lot receiving one or other firmware and/or model , the sound too , is much better on UV5R , i dunno why

**Comment by ErikS-web** [comment]
Interesting experiment

The crystal is 'far' away from the processor. There is a fairly long line leading to it. That alone is a huge antenna.

You can also try to make a shield over the processor and use a lot of ground points

**Reply by iu0azb** [reply]
It make sense to try indeed :) 
Would be also nice to have a full schematic of this radio.

**Comment by ErikS-web** [comment]
Perhabs this can help you

https://github.com/ludwich66/Quansheng_UV-K5_Wiki/wiki

and

https://github.com/amnemonic/Quansheng_UV-K5_Firmware/tree/main/hardware

**Reply by iu0azb** [reply]
Thanks Erik for  the links!!  Defenitly the TCXO is a good design choice.. I need to spend sometime to understand the configuration. 
I see that the decoupling cap is 1nF,  and 680ohm resistor is between the Xi and ground .. maybe to adapt the 10kohm tcxo output impedance?   From PCB detail I just noted a through hole between TCXO and "178" that seem to stay just under SW33, hope it doens't go anywere else :)

As previously discovered, BK datasheet unfortutately not mention any TCXO configuration.. that's a pity have no rules to implement from Beken.

By now easier next step is improve the shielding for that area ans see for any improvement.

**Comment by iu0azb** [comment]
Tonight I've crudely shielded all area over xtal and BK4819.. There are not too many spots where ground it easily. 
Not better result than shield just the xtal area.. The signal 130mhz signal on same test condition moved from -68 to -79dBm.  
I noticed that by bringing my finger closer to the EXIT key the signal increases dramatically... 
This gives me little hope for future developments....  Perhaps some daring experimenter or me in a lackluster moment I might think of halving the supply voltage of the TCXO :) 
![shield2](https://github.com/egzumer/uv-k5-firmware-custom/assets/63300383/ef63f4fa-bfe6-415b-a78a-ab5bb9f401ef)

**Reply by iu0azb** [reply]
Today I've noted that 26mhz harmonic signals are really weak now outside of the radio.. so at least the full shielding reduced the k5 emissions 😉
